src/main/scala/com/fulcrumgenomics/bam/CallOverlappingConsensusBases.scala

```diff
@@ -0,0 +1,182 @@
+    |   `--max-qual-on-agreement` is used.
+    |2. if the read and mate bases differ, then the base with the highest associated base quality will be the consensus
+    |   call.  If the read and mate have the same base quality, then the output base quality will be 2.  Otherwise,
+    |   the base quality will be the difference between the larger and smaller base quality.  The
+    |   `--only-mask-disagreements` option overrides behavior and sets all differing bases to `N` with a base quality of
+    |   2.
+    |
+    |The read and its mate will have their bases corrected and base qualities updated based on the procedure above.
+  """)
+class CallOverlappingConsensusBases
+( @arg(flag='i', doc="Input SAM or BAM file of aligned reads.") val input: PathToBam,
+  @arg(flag='o', doc="Output SAM or BAM file.") val output: PathToBam,
+  @arg(flag='m', doc="Output metrics file.") val metrics: FilePath,
+  @arg(doc="The number of threads to use while consensus calling.") val threads: Int = 1,
+  @arg(flag='S', doc="The sort order of the output. If not given, output will be in the same order as input if the input.")
+  val sortOrder: Option[SamOrder] = None,
+  @arg(doc="""If the read and mate bases disagree at a given reference position, true to mask (make 'N') the read and mate
+             |bases, otherwise pick the base with the highest base quality and return a base quality that's the difference
+             |between the higher and lower base qualities.""")
+  val onlyMaskDisagreements: Boolean = false,
+  @arg(doc= """If the read and mate bases agree at a given reference position, true to for the resulting base quality
+              |to be the maximum base quality, otherwise the sum of the base qualities.""")
+  val maxQualOnAgreement: Boolean = false
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  private case class ThreadData(caller: OverlappingBasesConsensusCaller, templateMetric: CallOverlappingConsensusBasesMetric, basesMetric: CallOverlappingConsensusBasesMetric)
+  private object ThreadData {
+    def apply(): ThreadData = ThreadData(
+      caller         = new OverlappingBasesConsensusCaller(onlyMaskDisagreements=onlyMaskDisagreements, maxQualOnAgreement=maxQualOnAgreement),
+      templateMetric = CallOverlappingConsensusBasesMetric(tpe=CountType.Templates),
+      basesMetric    = CallOverlappingConsensusBasesMetric(tpe=CountType.Bases)
+    )
+  }
+
+  override def execute(): Unit = {
+    val source           = SamSource(input)
+    val writer           = SamWriter(output, source.header)
+    val progress         = new ProgressLogger(logger)
+    val templateIterator = Bams.templateIterator(source)
+    val threadData       = new IterableThreadLocal(() => ThreadData())
+
+    ParIterator(templateIterator, threads=threads)
+      .map { template =>
+        val threadDatum = threadData.get()
+        threadDatum.synchronized {
```

Comment:
Why do you need synchronization here?  Aren't you only operating on a per-thread value?

Reply:
s